panic when register callback from go into python as module

code

```
package pythonbind

import "C"
import (
	"fmt"
	"github.com/sbinet/go-python"
	"time"
)

type Go2Python struct {
	fileName string
}

func Hello(self, args *python.PyObject) *python.PyObject {
	fmt.Printf("Hello world")
	return python.Py_None
}

// Run run the scripts
func (g Go2Python) Run() error {
	if err := python.Initialize(); err != nil {
		return err
	}
	methods := make([]python.PyMethodDef,1)
	method := &methods[0]
	method.Name = "hello"
	method.Meth = Hello
	method.Doc = "hello world"
	method.Flags = python.MethNoArgs

	_, err := python.Py_InitModule("snack", methods)
	if err != nil {
		panic(err)
	}
	if err := python.PyRun_SimpleFile(g.fileName); err != nil {
		return err
	}
	if err := python.Finalize(); err != nil {
		return err
	}
	time.Sleep(time.Minute * 5)
	return nil
}

```

when I run function run, panic happen in Py_InitModule

```
API server listening at: 127.0.0.1:60703
=== RUN   TestSum
--- FAIL: TestSum (7.60s)
panic: runtime error: cgo argument has Go pointer to Go pointer [recovered]
	panic: runtime error: cgo argument has Go pointer to Go pointer

goroutine 19 [running]:
testing.tRunner.func1(0xc0000e0100)
	/Users/xiyanxiyan10/go1.13/src/testing/testing.go:874 +0x60f
panic(0x4181560, 0xc0000984e0)
	/Users/xiyanxiyan10/go1.13/src/runtime/panic.go:679 +0x1e0
github.com/sbinet/go-python.cpyMethodDefs.func1(0x5003380, 0xc000049b78, 0xc0000ba080)
	/Users/xiyanxiyan10/project/goquant/back/pkg/mod/github.com/sbinet/go-python@v0.0.0-20190615090516-46d882be3991/heap.go:31 +0x12c
github.com/sbinet/go-python.cpyMethodDefs(0x41b0201, 0x5, 0xc000049e38, 0x1, 0x1, 0x0)
	/Users/xiyanxiyan10/project/goquant/back/pkg/mod/github.com/sbinet/go-python@v0.0.0-20190615090516-46d882be3991/heap.go:31 +0x2eb
github.com/sbinet/go-python.Py_InitModule(0x41b0201, 0x5, 0xc000049e38, 0x1, 0x1, 0x0, 0x0, 0x0)
	/Users/xiyanxiyan10/project/goquant/back/pkg/mod/github.com/sbinet/go-python@v0.0.0-20190615090516-46d882be3991/heap.go:42 +
```

go version 1.13

--- a/capi.go
+++ b/capi.go
@@ -16,10 +16,12 @@ func Py_BuildValue(format string, args ...interface{}) *PyObject {
 // ml_doc	char *	points to the contents of the docstring
 type PyMethodDef struct {
 	Name  string // name of the method
-	Meth  func(self, args *PyObject) *PyObject
+	Meth  PyCFunction
 	Flags MethodDefFlags
 	Doc   string
 }
+
+type PyCFunction C.PyCFunction
 
 type MethodDefFlags int
 

--- a/heap.go
+++ b/heap.go
@@ -24,7 +24,7 @@ func cpyMethodDefs(name string, methods []PyMethodDef) *C.PyMethodDef {
 	for i, meth := range methods {
 		cmeth := C.PyMethodDef{
 			ml_name:  C.CString(meth.Name),
-			ml_meth:  (C.PyCFunction)(unsafe.Pointer(&meth.Meth)),
+			ml_meth:  C.PyCFunction(meth.Meth),
 			ml_flags: C.int(meth.Flags),
 			ml_doc:   C.CString(meth.Doc),
 		}
@@ -43,6 +43,7 @@ func Py_InitModule(name string, methods []PyMethodDef) (*PyObject, error) {
 
 	obj := togo(C._gopy_InitModule(c_mname, cmeths))
 	if obj == nil {
+		PyErr_Print()
 		return nil, errors.New("python: internal error; module creation failed.")
 	}
 	return obj, nil
@@ -59,6 +60,7 @@ func Py_InitModule3(name string, methods []PyMethodDef, doc string) (*PyObject, 
 
 	obj := togo(C._gopy_InitModule3(cname, cmeths, cdoc))
 	if obj == nil {
+		PyErr_Print()
 		return nil, errors.New("python: internal error; module creation failed.")
 	}
 	return obj, nil

